klayout/drc/run_drc.py

```diff
@@ -0,0 +1,709 @@
+    layout_path : string
+        Path to the layout file that we will run DRC on.
+
+    Returns
+    -------
+    dict
+        Dictionary that represent all run switches passed to klayout.
+    """
+    switches = dict()
+
+    # No. of threads
+    thrCount = 2 if arguments["--thr"] is None else int(arguments["--thr"])
+    switches["thr"] = str(int(thrCount))
+
+    if arguments["--run_mode"] in ["flat", "deep", "tiling"]:
+        switches["run_mode"] = arguments["--run_mode"]
+    else:
+        logging.error("Allowed klayout modes are (flat , deep , tiling) only")
+        exit()
+
+    if arguments["--variant"] == "A":
+        switches["metal_top"] = "30K"
+        switches["mim_option"] = "A"
+        switches["metal_level"] = "3LM"
+        # switches = switches + f"-rd metal_top=30K -rd mim_option=A -rd metal_level=3LM "
+    elif arguments["--variant"] == "B":
+        switches["metal_top"] = "11K"
+        switches["mim_option"] = "B"
+        switches["metal_level"] = "4LM"
+        # switches = switches + f"-rd metal_top=11K -rd mim_option=B -rd metal_level=4LM "
+    elif arguments["--variant"] == "C":
+        switches["metal_top"] = "9K"
+        switches["mim_option"] = "B"
+        switches["metal_level"] = "5LM"
+        # switches = switches + f"-rd metal_top=9K  -rd mim_option=B -rd metal_level=5LM "
+    else:
+        logging.error("variant switch allowed values are (A , B, C) only")
+        exit()
+
+    if arguments["--verbose"]:
+        switches["verbose"] = "true"
+    else:
+        switches["verbose"] = "false"
+
+    if arguments["--no_feol"]:
+        switches["feol"] = "false"
+    else:
+        switches["feol"] = "true"
+
+    if arguments["--no_beol"]:
+        switches["beol"] = "false"
+    else:
+        switches["beol"] = "true"
+
+    if arguments["--no_offgrid"]:
+        switches["offgrid"] = "false"
+    else:
+        switches["offgrid"] = "true"
+
+    if arguments["--connectivity"]:
+        switches["conn_drc"] = "true"
+    else:
+        switches["conn_drc"] = "false"
+
+    if arguments["--density"]:
+        switches["density"] = "true"
+    else:
+        switches["density"] = "false"
+
+    switches["topcell"] = get_run_top_cell_name(arguments, layout_path)
+    switches["input"] = layout_path
+
+    return switches
+
+
+def check_klayout_version():
+    """
+    check_klayout_version checks klayout version and makes sure it would work with the DRC.
+    """
+    # ======= Checking Klayout version =======
+    klayout_v_ = os.popen("klayout -b -v").read()
+    klayout_v_ = klayout_v_.split("\n")[0]
+    klayout_v_list = []
+
+    if klayout_v_ == "":
+        logging.error("Klayout is not found. Please make sure klayout is installed.")
+        exit(1)
+    else:
+        klayout_v_list = [int(v) for v in klayout_v_.split(" ")[-1].split(".")]
+
+    logging.info(f"Your Klayout version is: {klayout_v_}")
+
+    if len(klayout_v_list) < 1 or len(klayout_v_list) > 3:
```

Comment:
should this be done before displaying the version?

Reply:
Done